src/axom/core/utilities/FileUtilities.hpp

```diff
@@ -119,6 +121,97 @@ void getDirName(std::string& dir, const std::string& path);
  */
 int removeFile(const std::string& filename);
 
+/**
+ * /brief Utility class for managing temporary files.
+ *
+ * The TempFile class provides a convenient way to create and manage temporary files.
+ * It ensures that the file is deleted upon destruction, if specified.
+ * The class is non-copyable to prevent accidental duplication of file handles.
+ *
+ * \note The path to the temp file is likely different from the supplied \a file_name
+ */
+class TempFile
+{
+public:
+  /**
+   * \brief Creates a tmp file using \a file_name and \a file_ext
+   * 
+   * \param file_name The name of the temporary file to create without extension
+   * \param file_ext An optional extension for the temp file
+   * \param delete_during_destruction If true (default), the file will be deleted when the TempFile object is destroyed.
```

Comment:
Question: If the file is not deleted when the object is destroyed, and the file name is randomly generated, how does one clean it up?

Reply:
Since we're not expecting to use this flag that often, I moved it out of the constructor and added a `retain(bool)` function.